docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,418 @@
+
+In the following example connecting `foo.a` to `bar.a`, whether `foo.a` is aligned with `foo` is irrelevant because the `:<>=` only computes alignment relative to the thing being connected to, and `foo.a` is aligned with `foo.a`.
+
+```scala mdoc:silent
+class Example1a extends RawModule {
+  val foo = IO(Flipped(new MixedAlignmentBundle))
+  val bar = IO(new MixedAlignmentBundle)
+  bar.a :<>= foo.a // whether foo.a is aligned/flipped to foo is IRRELEVANT to what gets connected with :<>=
+}
+```
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example1a())
+```
+
+While `foo.b`'s alignment with `foo` does not affect our operators, it does influence whether `foo.b` is an output or input port of my module.
+A common source of confusion is to mistake the process for determining whether `foo.b` is an output/input (the port-direction computation) with the process for determing how `:<>=` connects who to who (the connection-direction computation).
+While both processes consider relative alignment, they are distinct.
+
+The port-direction computation always computes alignment relative to the component marked with `IO`. An `IO(Flipped(gen))` is an input port, and any member of `gen` that is aligned/flipped with `gen` is an input/output port. An `IO(gen)` is an output port, and any member of `gen` that is aligned/flipped with `gen` is an output/input port.
+
+The connection-direction computation always computes alignment based on explicit consumer/producer referenced for the connection. If one connects `foo :<>= bar`, alignments are computed based on `foo` and `bar`. If I connect `foo.a :<>= bar.a`, then alignments are computed based on `foo.a` and `bar.a` (and the alignment of `foo` to `foo.a` is irrelevant).
+
+This means that users can try to assign to input ports of their module! If I write `x :<>= y`, and `x` is an input to my module, then that is what the connection is trying to do. However, because input ports are not assignable from within my module, Chisel will throw an error. This is the same error a user would get using a mono-directioned operator: `x := y` will throw the same error if `x` is an input module. *Whether a component is assignable is irrelevant to the semantics of any connection operator assigning to it.*
+
+In summary, the port-direction computation is relative to the root marked `IO`, but connection-direction computation is relative to the consumer/producer that the connection is doing. This has the positive property that connection semantics are solely based on the Chisel types of the consumer/producer (nothing more, nothing less).
+
+### Bi-direction connection operator (:<>=)
```

Comment:
shouldn't you have described this before the section above?

Reply:
I was torn which should go first... What do you think?

Reply:
maybe this first, and then the direction thing next?